app/models/manageiq/providers/ibm_power_hmc/inventory/collector/infra_manager.rb

```diff
@@ -1,68 +1,13 @@
 class ManageIQ::Providers::IbmPowerHmc::Inventory::Collector::InfraManager < ManageIQ::Providers::IbmPowerHmc::Inventory::Collector
-  def initialize(manager, target)
-    super
-    @netadapters = {}
-    @sriov_elps = {}
-    @vnics = {}
-    @ssps = {}
-    @pcm_enabled = {}
-  end
-
   def collect!
     $ibm_power_hmc_log.info("#{self.class}##{__method__}")
     manager.with_provider_connection do |connection|
-      @cecs = connection.managed_systems
-      do_lpars(connection)
-      do_vioses(connection)
-      do_vswitches(connection)
-      do_vlans(connection)
-      do_templates(connection)
-      do_ssps(connection)
-      do_pcm_preferences(connection)
-      $ibm_power_hmc_log.info("end collection")
-    rescue IbmPowerHmc::Connection::HttpError => e
-      $ibm_power_hmc_log.error("managed systems query failed: #{e}")
-      raise
+      @connection = connection
+      yield
```

Comment:
Is this yield buying us anything?  I think you could simply either:

```ruby
def initialize(manager, _target)
  super
  @connection = manager.connect
end
```

or

```ruby
def connection
  @connection ||= manager.connect
end
```

Reply:
No because we want to close the session with the HMC when we're done with the parsing (hence the with_provider_connection).

Reply:
Ah okay didn't realize you have an explicit logoff :+1: